src/mvba/vcbc/mod.rs

```diff
@@ -45,31 +46,93 @@ impl Vcbc {
 
         Self {
             ctx,
-            state: Box::new(ProposeState),
+            state: State::Propose,
         }
     }
 
     // propose sets the proposal and broadcast propose message.
     pub fn propose(&mut self, proposal: &Proposal) -> Result<()> {
-        debug_assert_eq!(proposal.proposer_id, self.ctx.proposer_id);
```

Comment:
I am not sure why you removed this debug assert. It's not bad to have it here.

Reply:
It was being check for below: https://github.com/maidsafe/sn_consensus/pull/70/files/8cc9e360b8c0ca75e655a7036a5b5561648ccaf1#diff-af91cddd2adf4b2869052ddc812b17294246da37b01643cdc33edfc550f7f04cR60

But yeah I could go either way, I don't mind keeping it